data-runtime/src/main/java/io/micronaut/data/runtime/mapper/BeanIntrospectionMapper.java

```diff
@@ -72,7 +69,7 @@ R map(@NonNull D object, @NonNull Class<R> type) throws InstantiationException {
                         } else {
                             Object convertFrom;
                             if (Collection.class.isAssignableFrom(argument.getType()) && !(o instanceof Collection)) {
-                                convertFrom = Collections.singleton(o);
+                                convertFrom = MapperUtils.toCollection(o);
```

Comment:
This was also not being done properly, needed to convert constructor arg to collection properly like the property in the code below. This is why I added record to test this behavior.